### DISCUSSION TITLE
How to import ditto into Eclipse?

### DISCUSSION THREAD

**Discussion by mohimoradi** [eclipse-ditto/ditto#1290] [Q&A]
I wrote an Eclipse plugin and am now collecting a database of large Java Maven projects. I tried to import ditto into Eclipse, but I couldn't compile it and run the tests.

I tried changing the java version or maven version, adding that plugin to maven dependencies, and following all of the StackOverflow suggestions, such as removing the.m2 folder, but none of them worked.

Is ditto Eclipse compatible? Is it possible to run their test classes inside Eclipse? If so, what procedure should I follow?

**Comment by thjaeckle** [comment]
To be honest: I did never try opening or compiling the Ditto codebase in the Eclipse IDE.
Ditto is a normal multi module maven project, which simply compiles using `mvn clean install`, so I don't see why an IDE should have problems with it.

What exactly is the issue? Do you get compiler errors?

**Reply by mohimoradi** [reply]
Thank you for your reply, Yes, I was able to compile ditto in the command prompt. But the tool is a JDT plugin that only runs in Eclipse IDE. I Just imported the ditto as an existing maven project in eclipse but it shows 1857 errors like the below samples: 

-AbstractDiagnosticLoggingAdapter cannot be resolved to a type	DefaultDittoDiagnosticLoggingAdapter.java	/ditto-internal-utils-akka/src/main/java/org/eclipse/ditto/internal/utils/akka/logging	line 212	Java Problem
-AkkaClassLoader cannot be resolved	QueryCriteriaValidator.java	/ditto-thingsearch-service/src/main/java/org/eclipse/ditto/thingsearch/service/persistence/query/validation	line 77	Java Problem
-The import org.eclipse.ditto.internal.utils.akka.logging.DittoDiagnosticLoggingAdapter cannot be resolved	AbstractHttpRequestActor.java	/ditto-gateway-service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/actors	line 62	Java Problem

![image](https://user-images.githubusercontent.com/81889496/150748704-e40d42ed-3526-4bdc-9203-b87377169542.png)

Ditto is a good Akka project and it is important for us to get it imported successfully into eclipse.

**Reply by thjaeckle** [reply]
Ah, based on the compile problems that could be an issue with the Scala code Ditto contains. Do you have a Scala (2.13) SDK setup/installed in your Eclipse IDE?

**Reply by mohimoradi** [reply]
Scala IDE(https://marketplace.eclipse.org/content/scala-ide) is no longer available to install. There are some unofficial versions but none of them work with the newer version of eclipse. I was able to successfully import sunbird-lms-service and Saga Actuator into eclipse. Is there any other way that helps me to compile ditto in eclipse?

Thank You!
